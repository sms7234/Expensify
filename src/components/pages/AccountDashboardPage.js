import React from 'react';
import AccountList from '../lists/AccountList';
import AccountSummary from '../independents/AccountSummary';
import AccountListFilters from '../filters/AccountListFilters';

const AccountDashboardPage = () => (
  <div>
    <AccountSummary />
    <AccountListFilters />
    <AccountList />
  </div>
)

export default AccountDashboardPage;
