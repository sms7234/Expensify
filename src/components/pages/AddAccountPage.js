import React from 'react';
import AccountForm from '../forms/AccountForm';
import {connect} from 'react-redux';
import {startAddAccount} from '../../actions/accounts';

export class AddAccountPage extends React.Component{
  onSubmit=(account) => {
    this.props.startAddAccount(account);
    this.props.history.push('/accounts');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title"> Add Account </h1>
          </div>
        </div>
        <div className="content-container">
          <AccountForm onSubmit={this.onSubmit} />
        </div>
      </div>
    );
  };
};

const mapDispatchToProps = (dispatch) => ({
  startAddAccount:(account) => dispatch(startAddAccount(account))
});

export default connect(undefined, mapDispatchToProps)(AddAccountPage);
