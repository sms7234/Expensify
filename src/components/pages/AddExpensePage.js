import React from 'react';
import ExpenseForm from '../forms/ExpenseForm';
import { connect} from 'react-redux';
import {startAddExpense} from '../../actions/expenses';

export class AddExpensePage extends React.Component{
  onSubmit = (expense) => {
    this.props.startAddExpense(expense);
    this.props.history.push('/expenses');
  };
  render(){
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title"> Add Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            onSubmit = {this.onSubmit}
            categoryList = {this.props.categoryList}
            accountList = {this.props.accountList}
            tagList = {this.props.tagList}
          />
        </div>
      </div>
    );
  }
};

const mapStateToProps=(state) => {
  return {
    accountList: state.accounts.map((item)=>{return({label: item.account, value: item.account})}),
    categoryList: state.categories.map((item)=>{return({label: item.category, value: item.category})}),
    tagList: state.tags.map((item)=>{return({label: item.tag, value: item.tag})})
  };
};

const mapDispatchToProps = (dispatch) => ({
  startAddExpense: (expense) => dispatch(startAddExpense(expense))
});

export default connect(mapStateToProps, mapDispatchToProps)(AddExpensePage);
