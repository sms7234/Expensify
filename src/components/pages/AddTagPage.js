import React from 'react';
import TagForm from '../forms/TagForm';
import {connect} from 'react-redux';
import {startAddTag} from '../../actions/tags';

export class AddTagPage extends React.Component{
  onSubmit=(tag) => {
    this.props.startAddTag(tag);
    this.props.history.push('/tags');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title"> Add Tag </h1>
          </div>
        </div>
        <div className="content-container">
          <TagForm onSubmit={this.onSubmit} />
        </div>
      </div>
    );
  };
};

const mapDispatchToProps = (dispatch) => ({
  startAddTag:(tag) => dispatch(startAddTag(tag))
});

export default connect(undefined, mapDispatchToProps)(AddTagPage);
