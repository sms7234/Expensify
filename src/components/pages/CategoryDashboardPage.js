import React from 'react';
import CategoryList from '../lists/CategoryList';
import CategoryListFilters from '../filters/CategoryListFilters';
import CategorySummary from '../independents/CategorySummary';

const CategoryDashboardPage = () => (
  <div>
    <CategorySummary />
    <CategoryListFilters />
    <CategoryList />
  </div>
)

export default CategoryDashboardPage;
