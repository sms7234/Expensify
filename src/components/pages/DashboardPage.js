import React from 'react';
import {connect} from 'react-redux';
import {Doughnut, Line} from 'react-chartjs-2';
import moment from 'moment';
import selectExpenses from '../../selectors/expenses'
import DashboardSummary from '../independents/DashboardSummary';
import ExpenseListFilters from '../filters/ExpenseListFilters';

export const DashboardPage = ({income, expenses, pieData, lineData, donutData}) => (
  <div>
    <DashboardSummary income={income} expenses={expenses} />
    <div className="content-container">
      <ExpenseListFilters />
      <div className="content-container--charts">
        <h1> Purchases by Category </h1>
        <Doughnut
          data={donutData}
        />
        <span />
        <h1> Remaining Funds (in $) by Date </h1>
        <Line
          data={lineData}
        />
      </div>
    </div>
  </div>
)

const mapStateToProps=(state)=>{
  //color array
  const colors = ['#E74C3C','#9B59B6', '#2980B9', '#1ABC9C', '#27AE60', '#F1C40F', '#F39C12', '#D35400', '#C0392B', '#8E44AD', '#3498DB', '#16A085', '#2ECC71', '#F5B041', '#EB984E' ];

  //assign colors - pass in array with desired length
  const colorAssign = (desiredQty) => {
    const max = colors.length;
    const holder = [];
    let n=0;
    desiredQty.forEach((item,index) => {
      if (index === max*(n+1)){n = n+1}
      holder.push(colors[index-(n*max)])
    })
    return holder;
  }

  //Sum data
  const sumItems = (arr) => {
    let holder = 0;
    if(arr!==[]){
      arr.forEach((item) => {
        holder = holder + item.amount
      })
    }
    return holder/100
  };

  //filter data
  const filterIncome = (arr) => {
    return arr.filter((item) => item.category==='Income')
  };

  const filterExpenses = (arr) => {
    return arr.filter((item)=> item.category!=="Income")
  };

  const FilterDoughnut = (arr) => {
    const labelHolder = {};

    //generate labels
    arr.forEach((item) => {
      if(!labelHolder.hasOwnProperty(item.category) && item.category!=="Income") {
        labelHolder[item.category]=0;
      }
    });
    //sum up values for each label
    arr.forEach((item) => {
      if(item.category !== "Income") {
        labelHolder[item.category] += item.amount/100;
      }
    })
    //generate colors
    const colorHolder = colorAssign(Object.keys(labelHolder));

    //reformat & returndata
    const donutData = {datasets:[{data: Object.values(labelHolder), backgroundColor: colorHolder}], labels:Object.keys(labelHolder)}
    return donutData;
  };

  const filterLineData = (arr) => {
    //holders
    const accounts = {};
    const labels= [];
    let counter = 0
    //sort by date
    const originalData = arr.sort((a,b) => {
      return a.purchaseDate < b.purchaseDate ? -1: 1;
    });
    //pull out unique dates & accounts
    originalData.forEach((item) => {
      const curDate = moment(item.purchaseDate).format('MM/DD/YY');
      if (!labels.includes(curDate)){
        labels.push(curDate)
      }
      if(!accounts.hasOwnProperty(item.account)) {
        accounts[item.account]=[0];
      }
    })
    //run through array & sort out expenses into acctObj
    let lastMatch = 0;
    const len = originalData.length;
    const accountList = Object.keys(accounts);
    labels.forEach((item,index) => {
      //add prev date sum (in accounts array) as the next item in array
      if(index>0){
        accountList.forEach((item) => {
          accounts[item].push(accounts[item][index-1]);
        })
      }
      for(let i=lastMatch; i<len; i++){
        //dates are the same...
        if(moment(item).isSame(originalData[i].purchaseDate, 'day')) {
          const acctName = originalData[i].account;
          if(originalData[i].category === 'Income'){
            accounts[acctName][index] += originalData[i].amount/100
          } else {
            accounts[acctName][index] -= originalData[i].amount/100
          }
          lastMatch = i;
          counter +=1;
        }
        //item date is greater than label date
        if(moment(originalData[i].purchaseDate).isAfter(item, 'day')) {
          break;
        }
      }
    })

    //assign colors
    const colorHolder = colorAssign(accountList);

    //reformat data for chart
    const datasets = [];
    accountList.forEach((item, index) => datasets.push({label: item, data: accounts[item], backgroundColor: colorHolder[index], borderColor: colorHolder[index], fill: false}))
    return {labels, datasets}

    //return correctly formatted information
    console.log('labels: ',labels, 'datasets: ',datasets)
    console.log('number of matches: ', counter, 'total expenses: ', originalData.length)

  }

  // actual data
    let visibleExpenses = selectExpenses(state.expenses, state.filters);
    let selectedExpenses = sumItems(filterExpenses(visibleExpenses));

  return {
    expenses: sumItems(filterExpenses(visibleExpenses)),
    income: sumItems(filterIncome(visibleExpenses)),
    lineData: filterLineData(selectExpenses(state.expenses, state.filters)),
    donutData: FilterDoughnut(selectExpenses(state.expenses, state.filters))
  };
};

export default connect(mapStateToProps)(DashboardPage);
