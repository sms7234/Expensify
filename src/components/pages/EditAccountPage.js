import React from 'react';
import {connect} from 'react-redux';
import AccountForm from '../forms/AccountForm';
import {startEditAccount, startRemoveAccount} from '../../actions/accounts';

export class EditAccountPage extends React.Component{
  onSubmit=(account)=>{
    this.props.startEditAccount(this.props.accounts.id, account),
    this.props.history.push('/accounts')
  };
  onRemove= ()=> {
    this.props.startRemoveAccount({id: this.props.accounts.id}),
    this.props.history.push('/accounts')
  };
  render(){
    return(
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title"> Edit Account</h1>
          </div>
        </div>
        <div className="content-container">
          <AccountForm
            accounts={this.props.accounts}
            onSubmit={this.onSubmit}
          />
          <button className="button--secondary" onClick={this.onRemove}>Remove Account</button>
        </div>
      </div>
    )
  }
};

const mapStateToProps = (state, props) => {
  return {
    accounts: state.accounts.find((item) => item.id === props.match.params.id)
  };
};

const mapDispatchToProps = (dispatch, props) => ({
  startEditAccount: (id,account) => dispatch(startEditAccount(id,account)),
  startRemoveAccount: (data) => dispatch(startRemoveAccount(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditAccountPage);
