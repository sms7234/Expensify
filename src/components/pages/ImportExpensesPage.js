import React from 'react';
import {connect} from 'react-redux';
import moment from 'moment';
import { v4 as uuidv4 } from 'uuid';
import {startAddExpense} from '../../actions/expenses';
import ImportInstructions from '../independents/ImportInstructions';
import ImportUploadExpenses from '../independents/ImportUploadExpenses';
import ImportExpenseListItem from '../items/ImportExpenseListItem';
import ImportSummary from '../independents/ImportSummary';


export class ImportExpensesPage extends React.Component {
  state= {
    data: [],
    validation: [],
    buttonSave: true,
    buttonCheck: true,
  };
  onDataUpload = (data, validation) => {
    this.setState(()=>({data, validation}));
    this.setState(()=>({
      buttonUpload: true,
      buttonCheck: false
    }));
  };
  onDataCheck = () => {
    const val= this.state.validation;
    const categories = [];
    const accounts = [];
    const tags = [];
    this.props.categoryList.forEach((item) => {
      categories.push(item.label);
    })
    this.props.accountList.forEach((item) => {
      accounts.push(item.label);
    })
    this.state.data.forEach((item,index) => {
      if(!item.Amount ||!item.Business ||!item.Category ||!item.Account){
        val[index]=false;
      } else if (item.Amount.length===0 || item.Business.length===0) {
        val[index]=false;
      } else if (!categories.includes(item.Category)) {
        val[index]=false;
      }else if (!accounts.includes(item.Account)) {
        val[index]=false;
      }
      else {
        val[index]=true;
      }
    });
    this.setState(()=>({validation: val}))
    if (val.includes(false) || val.includes(null)) {
      this.setState(()=>({
        buttonSave:true,
      }))
    } else {
      this.setState(()=>({buttonSave:false}))
    }
  };
  onDataAdd = () =>{
    const newData = {
      Date: moment(),
      Amount: '',
      Account: '',
      Category: '',
      Business: '',
      Tag:'',
      Note: ''
    };
    const data = this.state.data;
    const validation = this.state.validation;
    data.push(newData);
    validation.push(null);
    this.setState(()=>({data, validation, buttonCheck: false}))
  };
  onItemRemove = (index) => {
    const data = this.state.data;
    const validation = this.state.validation;
    data.splice(index,1);
    validation.splice(index,1);
    this.setState(()=>({data, validation}));
  };
  onItemSave=(update, index) => {
    const data = this.state.data;
    data[index] = update;
    this.setState(()=>({data}));
  };
  onSubmit=() =>{
    const convertedData = [];
    this.state.data.forEach((item)=>{
      convertedData.push({
        amount: parseInt(item.Amount)*100,
        business: item.Business,
        account: item.Account,
        tag: item.Tag,
        category: item.Category,
        note: item.Note,
        purchaseDate: item.Date.valueOf()
      })
    });
    convertedData.forEach((item) => {
      this.props.startAddExpense(item)
    })
    this.props.history.push('/expenses');
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title"> Import Expenses</h1>
            <h2 className="page-header__subtitle">Manually input or import csv data to add multiple expenses simultaneously</h2>
            <ImportSummary className="page-header__subtitle" qty={this.state.data} wordSingular={'expense'} wordPlural={'expenses'}/>
          </div>
        </div>
        <ImportInstructions
          dataFormat={['Date (MM/DD/YY)', 'Amount', 'Category', 'Business', 'Notes']}
        />
        <ImportUploadExpenses
          onDataUpload={this.onDataUpload}
          data={this.state.data}
          validation={this.state.validation}
        />
        <div className="content-container">
          <div className="list-header">
            <h4 className="show-for-mobile">Expense List </h4>
            <h4 className="show-for-desktop">Date</h4>
            <h4 className="show-for-desktop">Amount</h4>
            <h4 className="show-for-desktop">Category</h4>
            <h4 className="show-for-desktop">Business</h4>
            <h4 className="show-for-desktop">Account</h4>
            <h4 className="show-for-desktop">Tag</h4>
            <h4 className="show-for-desktop">Note</h4>
          </div>
          <div className="list-body">
          {
            !this.state.data ? (
              <div className="list-item--message">
                <span>No Data Available</span>
              </div>
            ): (
              this.state.data.map((item,index) => {
                return <ImportExpenseListItem
                  key = {uuidv4()}
                  {...item}
                  id={index}
                  dateKey={uuidv4()}
                  accountList={this.props.accountList}
                  categoryList={this.props.categoryList}
                  tagList={this.props.tagList}
                  validation={this.state.validation[index]}
                  onRemove={this.onItemRemove}
                  onSave={this.onItemSave}
                />;
              })
            )
          }
          </div>
          <div className="content-container--buttons">
            <button className="button" onClick={this.onDataAdd}>Add Data</button>
            <button className="button--check" disabled={this.state.buttonCheck} onClick={this.onDataCheck}>Check Data</button>
            <button className="button" disabled={this.state.buttonSave} onClick={this.onSubmit}>Submit Data</button>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    categoryList: state.categories.map((item)=>{return({label: item.category, value: item.category})}),
    accountList: state.accounts.map((item)=>{return({label: item.account, value: item.account})}),
    tagList: state.tags.map((item)=>{return({label: item.tag, value: item.tag})})
  };
};

const mapDispatchToProps = (dispatch) => ({
  startAddExpense: (expense) => dispatch(startAddExpense(expense))
});

export default connect(mapStateToProps, mapDispatchToProps)(ImportExpensesPage);
