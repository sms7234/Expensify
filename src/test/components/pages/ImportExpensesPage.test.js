import React from 'react';
import {shallow} from 'enzyme';
import moment from 'moment';
import {ImportExpensesPage} from '../../../components/pages/ImportExpensesPage';
import categoryList from '../../fixtures/categoryList';
import accountList from '../../fixtures/accountList';
import tagList from '../../fixtures/tagsList';

const fakeData = [
  {
    Amount: "5.00",
    Category: categoryList[0].value,
    Business: 'Walmart',
    Account: accountList[0].value,
    Tag: '',
    Date: moment(Date.parse('02/14/20')),
    Note: ''
  },
  {
    Amount: "25.00",
    Category: categoryList[1].value,
    Business: 'Chilis',
    Account: accountList[1].value,
    Tag: 'warranty',
    Date: moment(Date.parse('02/25/20')),
    Note: 'blah'
  }
];

const altData1=[
  {
    Amount: null,
    Category: null,
    Business: null,
    Account: null,
    Tag: null,
    Date: moment(),
    Note: ''
  }
];

const altData2=[
  {
    Amount: '',
    Category: '',
    Business: '',
    Account:'',
    Tag:'blah',
    Date: moment(),
    Note: ''
  }
];

test('render page with no data',() => {
  const wrapper = shallow(<ImportExpensesPage />);
  expect(wrapper).toMatchSnapshot();
});

test('test Add data button', () => {
  const newData = [{
    Date: moment(),
    Amount: '',
    Account:'',
    Tag:'',
    Category: '',
    Business: '',
    Note: ''
  }];
  const wrapper = shallow(<ImportExpensesPage categoryList={categoryList}/>);
  wrapper.find('button').at(0).simulate('click');
  expect(wrapper.state('data')).toEqual(newData);
});

test('test check button with correct data', () => {
  const wrapper = shallow(<ImportExpensesPage categoryList={categoryList} accountList={accountList}/>);
  wrapper.setState({data: fakeData});
  wrapper.setState({validation: [null, null]})
  wrapper.find('button').at(1).simulate('click');
  expect(wrapper.state('buttonSave')).toBe(false);
  const validation = wrapper.state('validation');
  expect(validation.includes(false)).toBe(false);
});

test('test check button with blank imported fields', () => {
  const wrapper = shallow(<ImportExpensesPage categoryList={categoryList} accountList={accountList}/>);
  wrapper.setState({data: altData1});
  wrapper.setState({validation: [null]})
  wrapper.find('button').at(1).simulate('click');
  expect(wrapper.state('buttonSave')).toBe(true);
  const validation = wrapper.state('validation');
  expect(validation.includes(false)).toBe(true);
})

test('test check button with incorrect amount data', () => {
  const wrapper = shallow(<ImportExpensesPage categoryList={categoryList} accountList={accountList}/>);
  wrapper.setState({data: altData2});
  wrapper.setState({validation: [null]})
  wrapper.find('button').at(1).simulate('click');
  expect(wrapper.state('buttonSave')).toBe(true);
  const validation = wrapper.state('validation');
  expect(validation.includes(false)).toBe(true);
})

test('test check button with incorrect business data', () => {
  const wrapper = shallow(<ImportExpensesPage categoryList={categoryList} accountList={accountList}/>);
  wrapper.setState({data: altData2});
  wrapper.setState({validation: [null]})
  wrapper.find('button').at(1).simulate('click');
  expect(wrapper.state('buttonSave')).toBe(true);
  const validation = wrapper.state('validation');
  expect(validation.includes(false)).toBe(true);
})

test('test check button with incorrect category data', () => {
  const wrapper = shallow(<ImportExpensesPage categoryList={categoryList} accountList={accountList}/>);
  wrapper.setState({data: altData2});
  wrapper.setState({validation: [null]})
  wrapper.find('button').at(1).simulate('click');
  expect(wrapper.state('buttonSave')).toBe(true);
  const validation = wrapper.state('validation');
  expect(validation.includes(false)).toBe(true);
});

test('test check button with incorrect account data', () => {
  const wrapper = shallow(<ImportExpensesPage categoryList={categoryList} accountList={accountList}/>);
  wrapper.setState({data: altData2});
  wrapper.setState({validation: [null]})
  wrapper.find('button').at(1).simulate('click');
  expect(wrapper.state('buttonSave')).toBe(true);
  const validation = wrapper.state('validation');
  expect(validation.includes(false)).toBe(true);
});

test('test check button with incorrect tag data', () => {
  const wrapper = shallow(<ImportExpensesPage categoryList={categoryList} accountList={accountList} tagList={tagList}/>);
  wrapper.setState({data: altData2});
  wrapper.setState({validation: [null]})
  wrapper.find('button').at(1).simulate('click');
  expect(wrapper.state('buttonSave')).toBe(true);
  const validation = wrapper.state('validation');
  expect(validation.includes(false)).toBe(true);
});

test('check that submit function runs', () => {
  const onSubmit = jest.fn()
  const history = {push: jest.fn()};
  const wrapper = shallow(<ImportExpensesPage categoryList={categoryList} startAddExpense={onSubmit} history={history} accountList={accountList}/>);
  wrapper.setState({data: fakeData});
  wrapper.find('button').at(2).simulate('click');
  expect(onSubmit).toHaveBeenCalled();
  expect(history.push).toHaveBeenLastCalledWith('/expenses');
})
