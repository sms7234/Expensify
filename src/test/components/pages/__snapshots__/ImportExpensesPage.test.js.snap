// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render page with no data 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
         Import Expenses
      </h1>
      <h2
        className="page-header__subtitle"
      >
        Manually input or import csv data to add multiple expenses simultaneously
      </h2>
      <ImportSummary
        className="page-header__subtitle"
        qty={Array []}
        wordPlural="expenses"
        wordSingular="expense"
      />
    </div>
  </div>
  <_default
    dataFormat={
      Array [
        "Date (MM/DD/YY)",
        "Amount",
        "Category",
        "Business",
        "Notes",
      ]
    }
  />
  <ImportUploadExpenses
    data={Array []}
    onDataUpload={[Function]}
    validation={Array []}
  />
  <div
    className="content-container"
  >
    <div
      className="list-header"
    >
      <h4
        className="show-for-mobile"
      >
        Expense List 
      </h4>
      <h4
        className="show-for-desktop"
      >
        Date
      </h4>
      <h4
        className="show-for-desktop"
      >
        Amount
      </h4>
      <h4
        className="show-for-desktop"
      >
        Category
      </h4>
      <h4
        className="show-for-desktop"
      >
        Business
      </h4>
      <h4
        className="show-for-desktop"
      >
        Account
      </h4>
      <h4
        className="show-for-desktop"
      >
        Tag
      </h4>
      <h4
        className="show-for-desktop"
      >
        Note
      </h4>
    </div>
    <div
      className="list-body"
    />
    <div
      className="content-container--buttons"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        Add Data
      </button>
      <button
        className="button--check"
        disabled={true}
        onClick={[Function]}
      >
        Check Data
      </button>
      <button
        className="button"
        disabled={true}
        onClick={[Function]}
      >
        Submit Data
      </button>
    </div>
  </div>
</div>
`;
